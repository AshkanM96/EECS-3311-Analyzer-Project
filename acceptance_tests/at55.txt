subtract
subtract
integer_constant(1)
integer_constant(2)
subtract
integer_constant(-5)
integer_constant(-10)
type_check
evaluate
