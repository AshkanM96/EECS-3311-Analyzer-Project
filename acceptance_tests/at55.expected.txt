  Expression currently specified: ?
  Report: Expression is initialized.
->subtract
  Expression currently specified: (? - nil)
  Report: OK.
->subtract
  Expression currently specified: ((? - nil) - nil)
  Report: OK.
->integer_constant(1)
  Expression currently specified: ((1 - ?) - nil)
  Report: OK.
->integer_constant(2)
  Expression currently specified: ((1 - 2) - ?)
  Report: OK.
->subtract
  Expression currently specified: ((1 - 2) - (? - nil))
  Report: OK.
->integer_constant(-5)
  Expression currently specified: ((1 - 2) - (-5 - ?))
  Report: OK.
->integer_constant(-10)
  Expression currently specified: ((1 - 2) - (-5 - -10))
  Report: OK.
->type_check
  Expression currently specified: ((1 - 2) - (-5 - -10))
  Report: ((1 - 2) - (-5 - -10)) is type-correct.
->evaluate
  Expression currently specified: ((1 - 2) - (-5 - -10))
  Report: -6
